### PR TITLE
fix(ui): forward modified keys (Alt / Ctrl) in focus-preview mode

### PR DESCRIPTION
**Pull request by seanmadden**
## Problem

In focus-preview mode (`Tab`), `handleFocusKey()` translates keypresses into `tmux send-keys` by switching on `msg.Type` only — it never looks at `msg.Alt`. So modified keys collapsed to their unmodified form:

- **Option/Alt+Backspace** → plain `BSpace` (deletes one char, not a word)
- **Alt+B / Alt+F / Alt+D** (word-back / word-fwd / kill-word) → a literal `b` / `f` / `d` typed into the pane
- **Ctrl+Left / Ctrl+Right** → fell through to `default:` and were sent as the literal string `ctrl+left`

Net effect: no word-wise line editing inside the focused pane (noticeable e.g. editing a long prompt through the preview).

## Fix

- Extract the keypress → `tmux send-keys` mapping into a pure `translateFocusKey(tea.KeyMsg) ([]focusKeySend, unfocus bool)` — no `Home`, no control client, no I/O — so it's unit-testable without a live `tmux -C`. `handleFocusKey()` now just dispatches its result.
- Forward `msg.Alt` keys with the tmux `M-` prefix: `M-BSpace`, `M-DC`, `M-Left/Right/Up/Down`, `M-Enter`, and `M-<rune>` for letters. Anything unmapped falls back to `Escape` then the bare key (Alt+X == ESC then X at the terminal level).
- Add explicit cases for `Ctrl+Left/Right/Up/Down` → `C-Left/…` and `Shift+Tab` → `BTab`.

## Tests

New `internal/ui/focuskey_test.go`:
- table test over plain keys, the new Ctrl/Shift mappings, and every Alt mapping;
- `TestTranslateFocusKey_ModifierNotDropped` — regression guard asserting `Alt+X` never produces the same send as plain `X` (the assertion that would have caught this bug);
- `TestTranslateFocusKey_AltFallbackEmitsEscape` — covers the `Escape`-then-key fallback.

`go build ./...`, `go vet ./...`, `go test ./internal/ui/...` all pass; `gofmt` clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved focus-mode key handling with enhanced support for Alt and Meta modifiers
  * Fixed Escape key behavior to properly exit focus mode and restore normal operations
  * Enhanced keyboard event forwarding and translation for improved tmux compatibility

* **Tests**
  * Added comprehensive test coverage for focus-mode key translation including modifier variants and edge cases

[Review Change Stack](https://app.coderabbit.ai/change-stack/brizzai/fleet/pull/73)

### CODE REVIEW

File: internal/ui/app.go [hunk 1]
```diff
@@ -1732,77 +1732,147 @@ func (h *Home) focusTick() tea.Cmd {
...
+	}
+
+	target := s.GetTmuxSession().Name
+	for _, sk := range sends {
+		if sk.literal {
+			cc.SendLiteralKeys(target, sk.val)
+		} else {
+			cc.SendKeys(target, sk.val)
 		}
 	}
```
[the remark this answers is not in the document]
Comment (seanmadden) [reply internal/ui/app.go:1908]: Keeping these sends synchronous in `Update()` is deliberate, and the new push adds a doc comment on `handleFocusKey` saying so. Two reasons:

1. **Ordering.** Bubble Tea runs `tea.Cmd`s on separate goroutines with no ordering guarantee. Wrapping each keystroke's sends in a `tea.Cmd` (as the suggestion does) means two fast keypresses — key-repeat, paste, holding Backspace — race, and characters can land out of order in the focused pane. Today, because `Update()` processes `KeyMsg`s sequentially on the main goroutine, forwarded keystrokes are in order. A correct async version needs an *ordered* send queue (single consumer goroutine on the `ControlClient`), which is a bigger change than this bug fix warrants.
2. **Cost.** Unlike the `git`/`gh` shell-outs that guideline targets, `ControlClient.SendKeys` is `fmt.Fprintf` of one line to the long-lived `tmux -C` PTY fd — ~0.1ms per the code's own comment, not a realistic Update-loop stall.

Note this isn't new in this PR — the old `handleFocusKey` already did the sends synchronously in the same spot; the refactor preserved it. Happy to do the ordered-queue version as a follow-up if you'd like it off the loop.

File: internal/ui/app.go [hunk 2]
```diff
@@ -1732,77 +1732,147 @@ func (h *Home) focusTick() tea.Cmd {
...
-		h.focusMode = false
-		h.sidebarDirty = true
-		return h, nil
+	if msg.Alt {
+		switch msg.Type {
+		case tea.KeyRunes:
+			for _, r := range msg.Runes {
+				sends = append(sends, focusKeySend{val: "M-" + string(r)})
+			}
+			return sends, false
```
[the remark this answers is not in the document]
Comment (seanmadden) [reply internal/ui/app.go]: Good catch — fixed in the latest push. The `Alt`+`KeyRunes` path now only emits `M-<rune>` when every rune is an ASCII letter (`allASCIILetters`); anything else (`;`, quotes, `#`, digits, …) falls back to `Escape` then the rune(s) sent via `-l` (which goes through `quoteTmux`), so arbitrary runes stay safely quoted. Covered by new table cases (`alt+semicolon`, `alt+digit`).

File: internal/ui/focuskey_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+		// Plain keys.
+		{"enter", tea.KeyMsg{Type: tea.KeyEnter}, []focusKeySend{{val: "Enter"}}, false},
+		{"backspace", tea.KeyMsg{Type: tea.KeyBackspace}, []focusKeySend{{val: "BSpace"}}, false},
+		{"tab", tea.KeyMsg{Type: tea.KeyTab}, []focusKeySend{{val: "Tab"}}, false},
+		{"shift+tab -> BTab", tea.KeyMsg{Type: tea.KeyShiftTab}, []focusKeySend{{val: "BTab"}}, false},
+		{"left", tea.KeyMsg{Type: tea.KeyLeft}, []focusKeySend{{val: "Left"}}, false},
+		{"ctrl+left -> C-Left", tea.KeyMsg{Type: tea.KeyCtrlLeft}, []focusKeySend{{val: "C-Left"}}, false},
+		{"ctrl+right -> C-Right", tea.KeyMsg{Type: tea.KeyCtrlRight}, []focusKeySend{{val: "C-Right"}}, false},
+		{"ctrl+w stays C-w", tea.KeyMsg{Type: tea.KeyCtrlW}, []focusKeySend{{val: "C-w"}}, false},
```
[the remark this answers is not in the document]
Comment (seanmadden) [reply internal/ui/focuskey_test.go:29]: Added — the table now covers `Ctrl+Up`/`Ctrl+Down` (→ `C-Up`/`C-Down`) and `Alt+Up`/`Alt+Down` (→ `M-Up`/`M-Down`) alongside the Left/Right variants.
